core/src/main/java/io/micronaut/core/annotation/SourceAnnotationValue.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.core.annotation;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.micronaut.core.convert.value.ConvertibleValues;
+import io.micronaut.core.util.ArgumentUtils;
+
+import java.lang.annotation.Annotation;
+import java.lang.annotation.RetentionPolicy;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * A runtime representation of the an stereotype annotation and its values.
```

Comment:
This javadoc isn't right. Should also mention how this only used at compile time